### ISSUE TITLE
Conflicting function names in apps

### ISSUE THREAD

**Issue by treeder** [fnproject/cli#39]
From @tteggel: 

If I have 4 funcs

```
app.yaml
a/create/func.yaml
a/delete/func.yaml
b/create/func.yaml
b/delete/func.yaml
```

with no routes in the func.yaml then I get routes for /create and /delete pointing to b's images

**Comment by treeder** [comment]
2 potential ways to fix this:

a) Fail if there’s name collisions (any function with the same name)
b) Prefix the name based on the directory structure, eg: a-create, b-create

Leaning towards b since it makes it easier for user. Maybe we just make the `name` field optional in func.yaml too, defaults to dir name. User can override the name, just as they can the path.

**Comment by treeder** [comment]
This will be fixed in #57 here: Closes #39 here: https://github.com/fnproject/cli/pull/57/files#diff-deb23646b3f0c40871119853be51362aR193

Takes path and puts dashes between the directories for the name.
